contents/docs/how-posthog-works/ingestion-pipeline.mdx

```diff
@@ -0,0 +1,172 @@
+
+###### Avoiding a merge
+
+Avoiding this scenario is the primary goal of the event buffer: by [buffering certain events](#1-event-buffer), we can avoid creating a duplicate Person when other events arrive before an `$identify` call.
+
+On the integration side, the best way to avoid this scenario is to send `$identify` events as early into a user's session as possible.
+This really only presents an issue when a user is logging in for the first time from a new device, and does not present any problems with users logging out and logging back in on an already seen device and platform.
+
+#### 1.2 - `$alias` events
+
+The process of handling `$alias` events is almost identical to the process for `$identify` events, except that instead of using `$distinct_id` and `$anon_distinct_id` as our two identifiers, we allow you to pass in two arbitrary `$distinct_id`'s you would like to combine.
+
+It's worth noting that in the case that both `$distinct_id`'s already have different Persons they are associated with, the `$distinct_id` that was provided second will be merged into the Person associated with the first `$distinct_id`
+
+#### 1.3 - All other events
+
+For all other types of events, the process is much more straightforward.
+If we have determined that this is a new `$distinct_id`, then we will create a new Person within PostgreSQL and associate them with this `$distinct_id`. Otherwise, we will retrieve the `person_id` associated with this `$distinct_id`, which will later be added as a field on the event in [Step 5](/docs/how-posthog-works/ingestion-pipeline#5-writing-to-clickhouse).
+
+#### 2. Update person properties
+
+Now, once we have finished determining the Person who is associated with the event we are processing, we can finish by updating their properties within PostgreSQL.
+This step takes into account any `$set`, `$set_once` or `$unset` arguments provided on the event, and merges these with any existing values for the Person.
+
+For more information on exactly how this updating is done, check out this overview of [user properties](/docs/integrate/user-properties).
+
+### 4. Event processing
+
+Finally, now that we have our event and person all ready, we perform a few last processing steps before we write the event to Clickhouse.
+This is our last chance to change anything about the event, which can include:
+
+-   Adding [group properties](/manual/group-analytics) if the event has been assigned to a Group
+-   Anonymizing IPs, if needed
+
+### 5. Writing to Clickhouse
+
+Now that we have our fully-processed event, the last step before we write to Clickhouse is to add the `person_id` and person properties to our event, both of which were determined in [Step 3](/docs/how-posthog-works/ingestion-pipeline#3-person-processing).
+Once we have all the person properties attached, we are ready to persist our event, which is done by sending it to a separate Kafka topic that Clickhouse will consume from and then write to its tables.
```

Comment:
I'd say we write the event and person id / person_props to the kafka topic (we don't really attach it to the event in any other sense)